### PR TITLE
Upgrade prio 0.14.1 -> 0.15.3 and fix related taskprov issues

### PR DESCRIPTION
**Pull request by cjpatton**
Based on #400 (merge that first).
Partially addresses #350.
Partially addresses #386.

Our taskprov implementation is meant to be compatible with DAP-02/VDAF-03. The latest version of the prio crate, 0.15.3, implements VDAF-07, so it's currently not operable. Taskprov will need to catch up with DAP-07/VDAF-07. In the meantime, remove support for the old VDAFs. So that we can keep the testing logic intact, add support for Prio2, which hasn't change over the same time period.

### CODE REVIEW

File: daphne/src/roles/mod.rs [hunk 1]
```diff
@@ -1918,11 +1918,11 @@ mod test {
             task_expiration: t.now + 86400 * 14,
             vdaf_config: taskprov::VdafConfig {
                 dp_config: taskprov::DpConfig::None,
-                var: taskprov::VdafTypeVar::Prio3Aes128Count,
+                var: taskprov::VdafTypeVar::Prio2 { dimension: 10 },
             },
         }
         .get_encoded_with_param(&t.helper.global_config.taskprov_version.unwrap());
-        let taskprov_id = crate::taskprov::compute_task_id(
```
Comment (cjpatton) [inline daphne/src/roles/mod.rs:1925]: My text editor highlights this in red, which suggests to me that `super` is preferred.

File: daphne/src/taskprov.rs [hunk 2]
```diff
@@ -334,14 +336,14 @@ mod test {
...
             &task_id,
-            VdafType::Prio3Aes128Count,
+            VdafType::Prio2,
         );
-        let expected: [u8; 16] = [
-            0xfb, 0xd1, 0x7d, 0xb5, 0x39, 0x0f, 0x94, 0x9e, 0xe3, 0x2d, 0x26, 0x34, 0xdc, 0x49,
-            0x9f, 0x5b,
+        let expected: [u8; 32] = [
+            251, 209, 125, 181, 57, 15, 148, 158, 227, 45, 38, 52, 220, 73, 159, 91, 145, 40, 123,
+            204, 49, 124, 7, 97, 221, 4, 232, 53, 194, 171, 19, 51,
```
Comment (cjpatton) [inline daphne/src/taskprov.rs:343]: @bhalleycf let me know if you prefer hex here.

**Review by mendess — APPROVED** [review]

File: daphne/src/taskprov.rs [hunk 3]
```diff
@@ -79,10 +78,10 @@ pub(crate) fn expand_prk_into_verify_key(
     // and it won't be, so we unwrap().
     let okm = prk.expand(&info, vdaf_type).unwrap();
     match &vdaf_type {
-        VdafType::Prio3Aes128Count | VdafType::Prio3Aes128Sum | VdafType::Prio3Aes128Histogram => {
-            let mut bytes = [0u8; 16];
+        VdafType::Prio2 => {
+            let mut bytes = [0u8; 32];
```
[the remark this answers is not in the document]
Comment (cjpatton) [reply daphne/src/taskprov.rs:82]: `32` needs to be `const`. I defined a constant for the key size and called it here and from `vdafy_type.len()`.

File: daphne/src/messages/taskprov.rs [hunk 4]
```diff
@@ -52,31 +43,21 @@ impl KeyType for VdafType {
...
-    // NOTE: this doesn't comply with the TaskProv spec, which doesn't match the VDAF-06 spec.
-    // Tracking the issue here: https://github.com/wangshan/draft-wang-ppm-dap-taskprov/issues/33.
-    Prio3Aes128Histogram { len: u32 },
-    Poplar1Aes128 { bit_length: u16 },
+    Prio2 {
+        dimension: u32,
+    },
+    #[cfg(test)]
     NotImplemented(u32),
 }
```
[the remark this answers is not in the document]
Comment (cjpatton) [reply daphne/src/messages/taskprov.rs:52]: Good idea, but I'll leave that refactoring for later.

**Pull request merged**